src/lib/registration/releasever.rb

```diff
@@ -0,0 +1,96 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+
+module Registration
+  # this class handles activating the new releasever value in the package management
+  class Releasever
+    include Yast::Logger
+
+    # the environment variable which overrides the system default in libzpp
+    RELEASEVER_ENV = "ZYPP_REPO_RELEASEVER"
+
+    Yast.import "Pkg"
+
+    # the new $releasever value
+    attr_reader :version
+
+    # constructor
+    # @param [String] version the new release version value
+    def initialize(version)
+      @version = version
+    end
+
+    # activate the new releasever
+    def activate
+      log.info "Setting $releasever to: #{version}"
+      # export the settings
+      ENV[RELEASEVER_ENV] = version
+
+      repos = repos_to_refresh
+      return if repos.empty?
+
+      # refresh the repositories and reload the packages
+      Yast::Pkg.SourceFinishAll
+      Yast::Pkg.SourceRestore
+      refresh(repos)
+      Yast::Pkg.SourceLoad
+    end
+
+    private
+
+    # get the repositories which contain "$releasever" directory in the URL
+    # @return [Array<Fixnum>] the repository list
+    def repos_to_refresh
+      repos = Yast::Pkg.SourceGetCurrent(true)
+      repos.select! do |repo|
+        # check the raw URL (without expanded variables)
+        raw_url = Yast::Pkg.SourceGeneralData(repo)["raw_url"]
+
+        next false unless raw_url
+
+        # The URL variables can be encoded is several ways, like $releasever or
```

Comment:
I'm not sure if this documentation will be available in the one generated by yardoc. I would be nice to have it available at class-level (or at least at method level). Just a suggestion.